picard/browser/browser.py

```diff
@@ -65,22 +83,45 @@ def stop(self):
 
     def _process_request(self):
         conn = self.sender()
-        line = bytes(conn.readLine()).decode()
-        conn.write(b"HTTP/1.1 200 OK\r\nCache-Control: max-age=0\r\n\r\nNothing to see here.")
-        conn.disconnectFromHost()
-        line = line.split()
-        log.debug("Browser integration request: %r", line)
-        if line[0] == "GET" and "?" in line[1]:
-            action, args = line[1].split("?")
-            args = [a.split("=", 1) for a in args.split("&")]
-            args = dict((a, QtCore.QUrl.fromPercentEncoding(b.encode('ascii'))) for (a, b) in args)
-            self.tagger.bring_tagger_front()
-            if action == "/openalbum":
-                self.tagger.load_album(args["id"])
-            elif action == "/opennat":
-                self.tagger.load_nat(args["id"])
+        rawline = conn.readLine().data()
+        log.debug("Browser integration request: %r", rawline)
+
+        def parse_line(line):
+            orig_line = line
+            try:
+                line = line.split()
+                if line[0] == "GET" and "?" in line[1]:
```

Comment:
This just seems to be doing a simple url parse. Why not use the python function instead of doing it manually? (I know you just moved this code, but just a suggestion)

Reply:
Done in faecb4cdd4b, not much shorter though, but still i think it's better.